src/vbd_store.ml

```diff
@@ -0,0 +1,37 @@
+open Lwt.Infix
+
+let m = Lwt_mutex.create ()
+
+let create_dir_if_doesnt_exist () =
+  Lwt.catch
+    (fun () -> Lwt_unix.mkdir Consts.xapi_nbd_persistent_dir 0o755)
+    (function
+      | Unix.(Unix_error (EEXIST, "mkdir", dir)) when dir = Consts.xapi_nbd_persistent_dir -> Lwt.return_unit
+      | e ->
+        Lwt_log.error_f "Failed to create directory: %s" (Printexc.to_string e)
```

Comment:
Now I see what you mean. I think if we have failures here, we should let the client fail and the user/admin fix the permission/space/... issue that caused the failure